onecodex/viz/_distance.py

```diff
@@ -1,118 +1,326 @@
+            in the resulting plot will change based on the metadata associated with each sample.
+        tooltip : `string` or `list`, optional
+            A string or list containing strings representing metadata fields. When a point in the
+            plot is hovered over, the value of the metadata associated with that sample will be
+            displayed in a modal.
+
+        Examples
+        --------
+        Scatter plot of weighted UniFrac distance between all our samples, using counts at the genus
+        level.
+
+        >>> plot_mds(rank='genus', metric='unifrac')
+
+        Notes
+        -----
+        **For `smacof`**: The values reported on the axis labels are Pearson's correlations between
+        the distances between points on each axis alone, and the corresponding distances in the
+        distance matrix calculated using the user-specified metric. These values are related to the
+        effectiveness of the MDS algorithm in placing points on the scatter plot in such a way that
+        they truly represent the calculated distances. They do not reflect how well the distance
+        metric captures similarities between the underlying data (in this case, an OTU table).
+        """
+        dists = self._compute_distance(rank, metric).to_data_frame()
+
+        # here we figure out what to put in the tooltips and get the appropriate data
+        if tooltip:
+            if not isinstance(tooltip, list):
+                tooltip = [tooltip]
+        else:
+            tooltip = []
+
+        tooltip = list(set(['Label', color, size] + tooltip))
+
+        magic_metadata, magic_fields = self._metadata_fetch(tooltip)
+
+        if method == 'smacof':
+            # adapted from https://scikit-learn.org/stable/auto_examples/manifold/plot_mds.html
+            x_field = 'MDS1'
+            y_field = 'MDS2'
+
+            seed = np.random.RandomState(seed=3)
+            mds = manifold.MDS(
+                max_iter=3000, eps=1e-12, random_state=seed, dissimilarity="precomputed", n_jobs=1
+            )
+            pos = mds.fit(dists).embedding_
+            plot_data = pd.DataFrame(pos, columns=[x_field, y_field], index=dists.index)
+            plot_data = plot_data.div(plot_data.abs().max(axis=0), axis=1)  # normalize to [0,1]
+
+            # determine how much of the original distance is captured by each of the axes after MDS.
+            # this implementation of MDS does not use eigen decomposition and so there's no simple
+            # way of returning a 'percent of variance explained' value
+            r_squared = []
+
+            for axis in [0, 1]:
+                mds_dist = pos.copy()
+                mds_dist[::, axis] = 0
+                mds_dist = squareform(euclidean_distances(mds_dist).round(6))
+                r_squared.append(pearsonr(mds_dist, squareform(dists))[0])
+
+            # label the axes
+            x_extra_label = 'r² = %.02f' % (r_squared[0],)
```

Comment:
I'm fine with the unicode, but could these be LaTeX instead? I think matplotlib can handle that, but not sure about other backends. This may prevent problems down the road with source file encodings.

Reply:
I wish =\ Altair's manual told me to just use unicode. Should we change it to `r-squared` instead?